sklearn/decomposition/sparse_pca.py

```diff
@@ -185,6 +191,63 @@ def transform(self, X, ridge_alpha='deprecated'):
         U /= s
         return U
 
+    def get_explained_variance(self, components, X):
+        '''
+            Get the explained variance from the principal components of the
+            data.
+
+            Parameters
+            ----------
+            components : 2D np.ndarray
+                the principal components of shape p x m
+                where m is the number of samples and p the number of
+                components
+            X : 2D np.ndarray
+                The feature vector. Must be shape m x n where m is the number
+                of samples and n is the dimension of the incoming feature
+                vector
+
+            Notes
+            -----
+            The variance ratio may not be computed. The main reason is that we
+            do not know what the total variance is since we did not compute all
+            the components.
+        '''
+        # requires the mean to be subtracted
+        X = X - np.mean(X, axis=0)
+        # the number of samples
+        n_samples = X.shape[0]
+        n_components = components.shape[0]
+        unit_vecs = components.copy()
+        unit_vecs = unit_vecs/np.linalg.norm(components, axis=1)[:, np.newaxis]
+
```

Comment:
if #11585 is merged, components will be already normalized.
However there will be a deprecation cycle for that, so for a few releases you'll have to check if the attribute introduced in that PR, `normalize`, is false before computing it

Reply:
ok, i think id rather wait and rebase, in case anything else changes, can you ping when done? (or feel free to take over if i take too long to respond!)

Reply:
will do :)

Reply:
I checked for `normalize_components` and if `False`, subtract mean and warn. Would that work?
thanks!